core/embed/rust/src/ui/model_mercury/component/hold_to_confirm.rs

```diff
@@ -0,0 +1,310 @@
+use crate::{
+    time::Duration,
+    ui::{
+        component::{Component, Event, EventCtx},
+        display::Color,
+        geometry::{Alignment2D, Offset, Rect},
+        lerp::Lerp,
+        shape,
+        shape::Renderer,
+    },
+};
+
+use super::{theme, Button, ButtonContent, ButtonMsg};
+
+#[cfg(feature = "haptic")]
+use crate::trezorhal::haptic::{self, HapticEffect};
+use crate::{
+    time::Stopwatch,
+    ui::{
+        component::Label,
+        geometry::{Alignment, Point},
+    },
+};
+use pareen;
+
+#[derive(Default, Clone)]
+struct HoldToConfirmAnim {
+    pub timer: Stopwatch,
+}
+
+impl HoldToConfirmAnim {
+    const DURATION: f32 = 2.2;
+
+    pub fn is_active(&self) -> bool {
+        self.timer.is_running_within(Duration::from(Self::DURATION))
+    }
+
+    pub fn eval(&self) -> (f32, f32, f32, f32, Color, f32, f32, f32) {
```

Comment:
this is very difficult to read. For now I'd at least document the intended usage.

Reply:
reorganized so that these anonymous return values are eliminated: https://github.com/trezor/trezor-firmware/pull/3776/commits/834ede0d3f37cd4d9d10e1d5a89626cef78385b6